### PR TITLE
C3 Custom Chart

### PR DESCRIPTION
**Pull request by vidma**
- [x] c3 chart. get autoresize for free (in report-view mode)
- [x] ability to pass optional `chartOptions`
- [x] some C3 charts are messed up (e.g. linechart)
  * looks like CSS conflict...

screen shot 2015-12-07 at 17 50 11

@andypetrella

### CODE REVIEW

File: notebooks/sql/DataFrame simple c3.snb [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
```
Comment (vidma) [inline notebooks/sql/DataFrame simple c3.snb]: this file will likely go to `viz/Custom charts with C3.snb`

File: modules/common/src/main/scala/notebook/front/widgets/package.scala [hunk 2]
```diff
@@ -506,6 +506,16 @@ package object widgets {
                                                   "width" → sizes._1, "height" → sizes._2)))
   }
 
+  case class CustomC3Chart[C:ToPoints:Sampler](originalData:C, chartOptions :String = "{}", override val sizes:(Int, Int)=(600, 400), maxPoints:Int = 25) extends Chart[C] {
```
Comment (andypetrella) [inline modules/common/src/main/scala/notebook/front/widgets/package.scala]: cool! however wouldn't be better to allow passing a `JsValue` that defaults to `Json.obj()`?

Comment (vidma) [inline modules/common/src/main/scala/notebook/front/widgets/package.scala]: not sure. 

- `JsValue` is Play-specific, so people would get scared ;) anyway they'd most likely [create it by parsing  regular json](https://www.playframework.com/documentation/2.2.x/ScalaJson#Using-string-parsing).
- maybe something similar could be useful for defining & combining a few preset `configurations`, like `stacked bar chart` with custom options like `column-name for x-axis`. But not convinced yet.

**Comment by vidma** [comment]
styles for some chart types are messed up, e.g. line chart (no clue how to fix this yet):

screen shot 2015-12-08 at 12 39 10

**Comment by vidma** [comment]
updated docs.
if not for issue with line charts, it is ready to :rocket:

**Comment by andypetrella** [comment]
it was a css clash right? I saw that in the pivot (I fixed the bar at that time I think)

**Comment by vidma** [comment]
> it was a css clash right? I saw that in the pivot (I fixed the bar at that time I think)

something like that, I just generalized and moved your fix into CSS, didn't go deep into details.

**Pull request merged**